app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -1,25 +1,74 @@
 package com.wire.android.ui
 
 import android.os.Bundle
-import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
 import androidx.compose.animation.ExperimentalAnimationApi
 import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import com.wire.android.navigation.NavigationGraph
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.navigation.navigateToItem
 import com.wire.android.ui.theme.WireTheme
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import javax.inject.Inject
 
 @ExperimentalMaterial3Api
 @ExperimentalAnimationApi
 @ExperimentalComposeUiApi
-class WireActivity : ComponentActivity() {
+@AndroidEntryPoint
+class WireActivity : AppCompatActivity() {
+
+    @Inject
+    lateinit var navigationManager: NavigationManager
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             WireTheme {
-                WireApp()
+
+                val scope = rememberCoroutineScope()
+                val navController = rememberNavController()
+
+                setUpNavigation(navController, scope)
+
+                Scaffold {
+                    NavigationGraph(navController = navController)
+                }
             }
         }
     }
+
+    @Composable
+    private fun setUpNavigation(
+        navController: NavHostController,
+        scope: CoroutineScope
+    ) {
+        // with the static key here we're sure that this effect wouldn't be canceled or restarted
+        LaunchedEffect("key") {
```

Comment:
What's this effect?

Reply:
Basically, to run suspended functions from Composable we need to use `LaunchedEffect` you can read more [here](https://developer.android.com/jetpack/compose/side-effects)